tools/stats/print_test_stats.py

```diff
@@ -675,20 +714,81 @@ def get_recursive_files(folder: str, extension: str) -> Iterable[str]:
+    num_unexpected_success = 0
+    num_errored = 0
+    num_skipped = 0
+    for test_run in runs:
+        if test_run.failed:
+            num_fail += 1
+        elif test_run.expected_failure:
+            num_expected_fail += 1
+        elif test_run.unexpected_success:
+            num_unexpected_success += 1
+        elif test_run.errored:
+            num_errored += 1
+        elif test_run.skipped:
+            num_skipped += 1
+        else:
+            num_pass += 1
+    err_msg = f'Warning: unintentional test case duplicates found for {test_run.name} in suite  {test_run.class_name}.'
+    report_only = os.getenv('PYTORCH_OVERRIDE_FLAKY_SIGNAL') != '1'
+    if report_only and num_fail + num_errored + num_unexpected_success < 1 or not report_only and num_expected_fail < 1:
+        raise RuntimeWarning(f'{err_msg} Intentional reruns are only triggered when the first run fails or errors, but'
+                             ' we found no failures nor errors.')
+    if num_unexpected_success + num_expected_fail < 1:
+        raise RuntimeWarning(f'{err_msg} Intentional reruns should raise at least one unexpected success or expected '
+                             'failure, but none have been found.')
+    if report_only and num_pass != num_unexpected_success:
+        raise RuntimeWarning(f'{err_msg} Every success in an intentional rerun is shadowed by one unexpected success.'
+                             f'However, successes = {num_pass} and unexpected successes = {num_unexpected_success}')
+    if not report_only and num_pass > 1:
+        raise RuntimeWarning(f'{err_msg} There should be at most 1 successful run in an intentional rerun that stops at'
+                             f' first success. The number of successful runs = {num_pass}')
+    if num_skipped > 0:
+        raise RuntimeWarning(f'{err_msg} No skips should occur in intentional reruns, but skips = {num_skipped}')
+    return max(num_unexpected_success, num_pass), num_fail + num_expected_fail + num_errored
+
+
+def assemble_flaky_test_stats(duplicated_tests_by_file: Dict[str, DuplicatedDict]) -> Any:
+    flaky_tests = []
+    for file_name, suite_to_dict in duplicated_tests_by_file.items():
+        for suite_name, testcase_to_runs in suite_to_dict.items():
+            for testcase_name, list_of_runs in testcase_to_runs.items():
+                num_green, num_red = process_intentional_test_runs(list_of_runs)
+                if num_green > 0:   # Otherwise, it's likely just a failing test
+                    flaky_tests.append({
+                        "name": testcase_name,
+                        "suite": suite_name,
+                        "file": file_name,
+                        "num_green": num_green,
+                        "num_red": num_red
+                    })
+    if len(flaky_tests) > 0:
+        register_rds_schema("flaky_tests", schema_from_sample(flaky_tests[0]))
+        rds_write("flaky_tests", flaky_tests, only_on_master=False)
```

Comment:
@driazati Is this the proper way to write to Grafana?

Reply:
yup looks good to me, you should see a `flaky` tests table show up [here](https://metrics.pytorch.org/explore?orgId=1&left=%5B%22now-1h%22,%22now%22,%22Metrics%22,%7B%22datasource%22:%22Metrics%22,%22format%22:%22table%22,%22timeColumn%22:%22sync_last_update_at%22,%22metricColumn%22:%22none%22,%22group%22:%5B%5D,%22where%22:%5B%7B%22type%22:%22macro%22,%22name%22:%22$__timeFilter%22,%22params%22:%5B%5D%7D%5D,%22select%22:%5B%5B%7B%22type%22:%22column%22,%22params%22:%5B%22pk_id%22%5D%7D%5D%5D,%22rawQuery%22:true,%22rawSql%22:%22show%20tables;%22,%22table%22:%22pull_request_review_event%22,%22timeColumnType%22:%22timestamp%22%7D%5D) once this runs in GHA (you can also get AWS credentials and set them in your env to run it locally)